app/com/m3/octoparts/aggregator/handler/HttpPartRequestHandler.scala

```diff
@@ -98,7 +98,7 @@ trait HttpPartRequestHandler extends Handler {
       AggregateRequestIdHeader -> partRequestInfo.requestMeta.id,
       PartRequestIdHeader -> partRequestInfo.partRequestId,
       PartIdHeader -> partRequestInfo.partRequest.partId
-    )
+    ) ++ partRequestInfo.requestMeta.proxyId.map { case s => ProxyIdHeader -> s }
```

Comment:
I don't think we need a `case` here.

We should also make sure to test that the header is being properly forwarded (and left out if no proxy) in subsequent requests to backends from Octoparts.

Also, since this method is called `buildTracingHeaders`, we should probably do proxy id header adding in another method or rename this method.